test/unit/test_model_server.py

```diff
@@ -0,0 +1,330 @@
+@patch("os.path.exists", return_value=True)
+@patch("sagemaker_pytorch_serving_container.torchserve._create_torchserve_config_file")
+@patch("sagemaker_pytorch_serving_container.torchserve._adapt_to_ts_format")
+def test_start_torchserve_default_service_handler(
+    adapt,
+    create_config,
+    exists,
+    install_requirements,
+    sigterm,
+    retrieve,
+    subprocess_popen,
+    subprocess_call,
+):
+    torchserve.start_torchserve()
+
+    adapt.assert_called_once_with(torchserve.DEFAULT_HANDLER_SERVICE)
+    create_config.assert_called_once_with()
+    exists.assert_called_once_with(REQUIREMENTS_PATH)
+    install_requirements.assert_called_once_with()
+
+    ts_model_server_cmd = [
+        "torchserve",
+        "--start",
+        "--model-store",
+        torchserve.MODEL_STORE,
+        "--ts-config",
+        torchserve.TS_CONFIG_FILE,
+        "--log-config",
+        torchserve.DEFAULT_TS_LOG_FILE,
+        "--models",
+        "model.mar"
+    ]
+
+    subprocess_popen.assert_called_once_with(ts_model_server_cmd)
+    sigterm.assert_called_once_with(retrieve.return_value)
+
+
+@patch("subprocess.call")
+@patch("subprocess.Popen")
+@patch("sagemaker_pytorch_serving_container.torchserve._retrieve_ts_server_process")
+@patch("sagemaker_pytorch_serving_container.torchserve._add_sigterm_handler")
+@patch("sagemaker_pytorch_serving_container.torchserve._install_requirements")
+@patch("os.path.exists", return_value=True)
+@patch("sagemaker_pytorch_serving_container.torchserve._create_torchserve_config_file")
+@patch("sagemaker_pytorch_serving_container.torchserve._adapt_to_ts_format")
+def test_start_torchserve_default_service_handler_multi_model(
+    adapt,
+    create_config,
+    exists,
+    install_requirements,
+    sigterm,
+    retrieve,
+    subprocess_popen,
+    subprocess_call,
+):
+    torchserve.ENABLE_MULTI_MODEL = True
+    torchserve.start_torchserve()
+    torchserve.ENABLE_MULTI_MODEL = False
+    create_config.assert_called_once_with()
+    exists.assert_called_once_with(REQUIREMENTS_PATH)
+    install_requirements.assert_called_once_with()
+
+    ts_model_server_cmd = [
+        "torchserve",
+        "--start",
+        "--model-store",
+        torchserve.MODEL_STORE,
+        "--ts-config",
+        torchserve.TS_CONFIG_FILE,
+        "--log-config",
+        torchserve.DEFAULT_TS_LOG_FILE,
+        "--models",
+        "model.mar"
+    ]
+
+    subprocess_popen.assert_called_once_with(ts_model_server_cmd)
+    sigterm.assert_called_once_with(retrieve.return_value)
+
+
+@patch("subprocess.call")
+@patch("subprocess.Popen")
+@patch("sagemaker_pytorch_serving_container.torchserve._retrieve_ts_server_process")
+@patch("sagemaker_pytorch_serving_container.torchserve._add_sigterm_handler")
+@patch("sagemaker_pytorch_serving_container.torchserve._create_torchserve_config_file")
+@patch("sagemaker_pytorch_serving_container.torchserve._adapt_to_ts_format")
+def test_start_torchserve_custom_handler_service(
```

Comment:
is the multi-model case also tested?

Reply:
Yes. You should see torchserve.ENABLE_MULTI_MODEL = True / torchserve.ENABLE_MULTI_MODEL = False in the test code. 

These are there to test the multi model case